app/scripts/modules/kubernetes/src/common/tolerations/tolerations.component.ts

```diff
@@ -0,0 +1,108 @@
+import { IComponentOptions, IController, module } from 'angular';
+
+class KubernetesTolerationsController implements IController {
+  public tolerations: any;
+  private operators = ['Exists', 'Equal'];
+  private effects = ['NoSchedule', 'PreferNoSchedule', 'NoExecute'];
+
+  public addToleration() {
+    if (!this.tolerations) {
+      this.tolerations = [];
+    }
+
+    this.tolerations.push({
+      key: '',
+      operator: 'Equal',
+      effect: 'NoSchedule',
+      value: '',
+    });
+  }
+
+  public removeToleration(index: any) {
+    this.tolerations.splice(index, 1);
+  }
+}
+
+class KubernetesTolerationsComponent implements IComponentOptions {
+  public bindings: any = { tolerations: '=' };
+  public controller: any = KubernetesTolerationsController;
+  public controllerAs: any = 'ctrl';
```

Comment:
Can omit `any`, `controllerAs` will be implicitly typed as a `string` from the assignment.